tests/foreman/cli/test_repository.py

```diff
@@ -585,12 +585,13 @@ def test_negative_create_non_yum_with_download_policy(self):
         @Assert: Non-YUM repository is not created with a download policy
         """
         for content_type in REPO_TYPE.keys():
-            if content_type != 'yum':
-                with self.assertRaises(CLIFactoryError):
-                    self._make_repository({
-                        u'content-type': content_type,
-                        u'download-policy': u'on_demand'
-                    })
+            with self.subTest(content_type):
+                if content_type != 'yum':
```

Comment:
Here you can filter out 'yum' from REPO_TYPe before entering the loop.

```python
non_yum_repo_types = [item for item in REPO_TYPE.keys() if item != 'yum']
for content_type in non_yum_repo_types:
    with self.subTest(content_type):
        with self.assertRaises(HTTPError):
```
The **small** advantage of this is that you avoid the use of one nested if and also avoids the creation of one useless **subTest**